### PR TITLE
Solr 4th wave

### PR DESCRIPTION
**Pull request by Nunien**
- minimum requests
- facet paging
- facet sorting

Note this limitations : 
    - Solr only allows to sort by index (alphabetically) or by count
    - In sort by count mode, selecting a facet value changes the order. That's why, in this case, the list automatically goes to page 1
    - Solr don't allow to know in advance the total number of facets when paging, thus we cannot know the number of pages

### CODE REVIEW

**Comment by jcomedouteau** [comment]
Tell me when I can test !

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/solr/repository/SolrRepositoryImpl.java [hunk 1]
```diff
@@ -32,147 +55,292 @@
...
-		if (facet.getDatasetEndDate() != null) {
-			criteria.and(Criteria.where("datasetCreationDate").lessThanEqual(DateTimeUtils.localDateToSolrString(facet.getDatasetEndDate())));
+		
+		/* configure the returned facet values */
+		addFacetPaging(query, shanoirQuery);
+		
+		/* query Solr server */
+		QueryResponse response;
+		try {
+			LOG.debug("Solr search : " + query);
```
[the remark this answers is not in the document]
Comment (Nunien) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/solr/repository/SolrRepositoryImpl.java:203]: I think the debug level is appropriate.

[the remark this answers is not in the document]
Comment (jcomedouteau) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/solr/repository/SolrRepositoryImpl.java:203]: ok !

**Comment by jcomedouteau** [comment]
Hi julien
One subject of discussion: Do we filter by default by alphabetical order or by number of results ?

I select a study -> I want them to appear by alphabetical order
I want the subject of the concerned study to appear first, so I want them to appear by number of result.

Could we imagine "when we have at least one facet selected, then order by number of results" ?

Otherwise, everything is good for me :)

**Review by jcomedouteau — APPROVED** [review]

**Pull request merged**